api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +256,85 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' concept values.
+	 * 
+	 * <ol>
+	 *     <li>Validates if high absolute and low absolute are within the valid range</li>
+	 *     <li>Validates if patient's age is within the valid range</li>
+	 * <ol/>
+	 * in the {@link ConceptReferenceRange}.
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 *               
+	 * @since 2.7.0
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		Concept concept = obs.getConcept();
+
+		if (concept != null && concept.getDatatype() != null && concept.getDatatype().isNumeric()) {
+			List<ConceptReferenceRange> crrList = Context.getConceptService()
+				.getConceptReferenceRangesByConceptId(concept.getConceptId());
+
+			if (!crrList.isEmpty()) {
+				ConceptReferenceRangeUtility utility = new ConceptReferenceRangeUtility();
+				boolean criteriaEvaluated = false;
+
+				for (ConceptReferenceRange referenceRange : crrList) {
+					if (utility.evaluateCriteria(referenceRange.getCriteria(), obs.getPerson())) {
+						criteriaEvaluated = true;
+						validateAbsoluteRanges(obs, referenceRange, errors);
+						setObsReferenceRange(obs, referenceRange);
+						break;
+					}
+				}
+
+				if (!criteriaEvaluated) {
```

Comment:
Why are we rejecting the value when no criteria evaluated to true?

Reply:
This means no criteria that would fit the obs's attributes e.g. patient's age could be out of range.

Reply:
Would that mean rejecting the obs?

Reply:
Yes, or do we want to continue with Obs creation if no criteria evaluated to true? I am okay with this as well

Reply:
Rejecting a value which did not fail any validation does not seem like what we should do. Don't you think so? 😊

Reply:
Right. However, I still have a concern around this. Say we have criteria = "$person.getAge() <= 3" to evaluate, and the evaluation returns false (fails), say because the age of the patient in subject is 5 years. Then we will proceed to save the Obs with wrong/unwanted values

Reply:
My understanding is that when one sets the criteria for only those below 3years, they mean that for those above 3 years, there is no need of setting their criteria because the default values which are already in the concept_numeric table takes care of that. Not so? 😊

Reply:
Actually it makes sense because if the criteria evaluates to true, then we want to evaluate absolute values (which are critical, say for that age), otherwise we don't need to check the ranges if criteria did not evaluate to true. I will remove this check.